File: CSharp/GazaSort.cs
using System;
using System.Collections.Generic;

public class GazaSort {
    public static List<int> Sort(List<int> arr) {
        Random rnd = new Random();
        while (!IsSorted(arr)) {
            arr.RemoveAt(rnd.Next(arr.Count));
        }
        return arr;
    }

    private static bool IsSorted(List<int> arr) {
        for (int i = 1; i < arr.Count; i++) {
            if (arr[i - 1] > arr[i]) {
                return false;
            }
        }
        return true;
    }
}

File: PHP/gazasort.php
function gazaSort(array $arr): array {
    while (!isSorted($arr)) {
        array_splice($arr, rand(0, count($arr) - 1), 1);
    }
    return $arr;
}

function isSorted(array $arr): bool {
    for ($i = 1; $i < count($arr); $i++) {
        if ($arr[$i - 1] > $arr[$i]) {
            return false;
        }
    }
    return true;
}
